It looks like the build failure is not caused by this PR, but because emacs is broken in master.

<details><summary>result of nixpkgs-review pr</summary>
<pre>
<code>
$ nix run -- nixpkgs#nixpkgs-review pr --build-args="--print-build-logs" https://github.com/NixOS/nixpkgs/pull/252562
$ git -c fetch.prune=false fetch --no-tags --force https://github.com/NixOS/nixpkgs master:refs/nixpkgs-review/0 pull/252562/head:refs/nixpkgs-review/1
$ git worktree add /home/lelgenio/.cache/nixpkgs-review/pr-252562-3/nixpkgs 78ec795ab148db73859f95f55306082de727d1f3
Preparing worktree (detached HEAD 78ec795ab148)
Updating files: 100% (37064/37064), done.
HEAD is now at 78ec795ab148 Merge pull request #252551 from r-ryantm/auto-update/discord
$ git merge --no-commit --no-ff febd84ace449485c2ee163b969c5298626e1f990
Automatic merge went well; stopped before committing as requested
$ nix build --extra-experimental-features nix-command no-url-literals --no-link --keep-going --no-allow-import-from-derivation --option build-use-sandbox relaxed -f /home/lelgenio/.cache/nixpkgs-review/pr-252562-3/build.nix --print-build-logs
emacs> unpacking sources
emacs> unpacking source archive /nix/store/833xz3irmdc8c87pdda1x49nn3lv5ys1-source
emacs> source root is source
emacs> patching sources
emacs> applying patch /nix/store/08jccfyhhyb0lz1yhns41pnlw9461bpg-native-comp-driver-options-28.patch
emacs> patching file lisp/emacs-lisp/comp.el
emacs> substituteStream(): WARNING: pattern '/bin/pwd' doesn't match anything in file './admin/charsets/Makefile.in'
emacs> substituteStream(): WARNING: pattern '/bin/pwd' doesn't match anything in file './admin/grammars/Makefile.in'
emacs> substituteStream(): WARNING: pattern '/bin/pwd' doesn't match anything in file './admin/unidata/Makefile.in'
emacs> substituteStream(): WARNING: pattern '/bin/pwd' doesn't match anything in file './doc/emacs/Makefile.in'
emacs> substituteStream(): WARNING: pattern '/bin/pwd' doesn't match anything in file './doc/lispintro/Makefile.in'
emacs> substituteStream(): WARNING: pattern '/bin/pwd' doesn't match anything in file './doc/lispref/Makefile.in'
emacs> substituteStream(): WARNING: pattern '/bin/pwd' doesn't match anything in file './doc/misc/Makefile.in'
emacs> substituteStream(): WARNING: pattern '/bin/pwd' doesn't match anything in file './leim/Makefile.in'
emacs> substituteStream(): WARNING: pattern '/bin/pwd' doesn't match anything in file './lib/Makefile.in'
emacs> substituteStream(): WARNING: pattern '/bin/pwd' doesn't match anything in file './lisp/Makefile.in'
emacs> substituteStream(): WARNING: pattern '/bin/pwd' doesn't match anything in file './lwlib/Makefile.in'
emacs> substituteStream(): WARNING: pattern '/bin/pwd' doesn't match anything in file './msdos/autogen/Makefile.in'
emacs> substituteStream(): WARNING: pattern '/bin/pwd' doesn't match anything in file './oldXMenu/Makefile.in'
emacs> substituteStream(): WARNING: pattern '/bin/pwd' doesn't match anything in file './src/Makefile.in'
emacs> substituteStream(): WARNING: pattern '/bin/pwd' doesn't match anything in file './test/Makefile.in'
emacs> autoreconfPhase
emacs> autoreconf: export WARNINGS=
emacs> autoreconf: Entering directory '.'
emacs> autoreconf: configure.ac: not using Gettext
emacs> autoreconf: running: aclocal --force
emacs> autoreconf: configure.ac: tracing
emacs> autoreconf: configure.ac: not using Libtool
emacs> autoreconf: configure.ac: not using Intltool
emacs> autoreconf: configure.ac: not using Gtkdoc
emacs> autoreconf: running: /nix/store/3zhnqp7v429h6s1vdpsx6v5v26jb4bs6-autoconf-2.71/bin/autoconf --force
emacs> autoreconf: running: /nix/store/3zhnqp7v429h6s1vdpsx6v5v26jb4bs6-autoconf-2.71/bin/autoheader --force
emacs> autoreconf: configure.ac: not using Automake
emacs> autoreconf: 'build-aux/config.sub' is updated
emacs> autoreconf: 'build-aux/config.guess' is updated
emacs> autoreconf: Leaving directory '.'
emacs> updateAutotoolsGnuConfigScriptsPhase
emacs> Updating Autotools / GNU config script to a newer upstream version: ./build-aux/config.sub
emacs> Updating Autotools / GNU config script to a newer upstream version: ./build-aux/config.guess
emacs> configuring
emacs> configure flags: --prefix=/nix/store/5jjzskl9ai74zmcn9xncckc7vhina811-emacs-28.2 --disable-build-details --with-modules --with-x-toolkit=lucid --with-xft --with-cairo --with-native-compilation
emacs> checking for xcrun... no
emacs> checking for GNU Make... make
emacs> checking build system type... x86_64-pc-linux-gnu
emacs> checking host system type... x86_64-pc-linux-gnu
emacs> checking whether the C compiler works... yes
emacs> checking for C compiler default output file name... a.out
emacs> checking for suffix of executables...
emacs> checking whether we are cross compiling... no
emacs> checking for suffix of object files... o
emacs> checking whether the compiler supports GNU C... yes
emacs> checking whether gcc accepts -g... yes
emacs> checking for gcc option to enable C11 features... none needed
emacs> checking whether the compiler is clang... no
emacs> checking for compiler option needed when checking for declarations... none
emacs> checking whether gcc and cc understand -c and -o together... yes
emacs> checking for stdio.h... yes
emacs> checking for stdlib.h... yes
emacs> checking for string.h... yes
emacs> checking for inttypes.h... yes
emacs> checking for stdint.h... yes
emacs> checking for strings.h... yes
emacs> checking for sys/stat.h... yes
emacs> checking for sys/types.h... yes
emacs> checking for unistd.h... yes
emacs> checking for wchar.h... yes
emacs> checking for minix/config.h... no
emacs> checking for linux/fs.h... yes
emacs> checking for malloc.h... yes
emacs> checking for sys/systeminfo.h... no
emacs> checking for sys/sysinfo.h... yes
emacs> checking for coff.h... no
emacs> checking for pty.h... yes
emacs> checking for sys/resource.h... yes
emacs> checking for sys/utsname.h... yes
emacs> checking for pwd.h... yes
emacs> checking for utmp.h... yes
emacs> checking for util.h... no
emacs> checking for sanitizer/lsan_interface.h... yes
emacs> checking for sys/socket.h... yes
emacs> checking for sys/param.h... yes
emacs> checking for pthread.h... yes
emacs> checking for malloc/malloc.h... no
emacs> checking for sys/un.h... yes
emacs> checking for vfork.h... no
emacs> checking for dirent.h... yes
emacs> checking for execinfo.h... yes
emacs> checking for stdio_ext.h... yes
emacs> checking for sys/vfs.h... yes
emacs> checking for sys/fs_types.h... no
emacs> checking for getopt.h... yes
emacs> checking for sys/cdefs.h... yes
emacs> checking for sys/time.h... yes
emacs> checking for ieee754.h... yes
emacs> checking for limits.h... yes
emacs> checking for sys/select.h... yes
emacs> checking for sys/random.h... yes
emacs> checking whether it is safe to define __EXTENSIONS__... yes
emacs> checking whether _XOPEN_SOURCE should be defined... no
emacs> checking how to run the C preprocessor... gcc -E
emacs> checking for grep that handles long lines and -e... /nix/store/m88w8in2zf5dwijxypyba02f0ff260rm-gnugrep-3.11/bin/grep
emacs> checking for egrep... /nix/store/m88w8in2zf5dwijxypyba02f0ff260rm-gnugrep-3.11/bin/grep -E
emacs> checking for Minix Amsterdam compiler... no
emacs> checking for ar... ar
emacs> checking for special C compiler options needed for large files... no
emacs> checking for _FILE_OFFSET_BITS value needed for large files... no
emacs> checking whether gcc accepts -g3 -O2... yes
emacs> checking whether the compiler is clang... no
emacs> checking whether C compiler handles -Werror -Wunknown-warning-option... no
emacs> checking for a BSD-compatible install... /nix/store/5wspmqhgsfs7dza9gcyrj5yxbyd00q7y-coreutils-9.3/bin/install -c
emacs> checking command to symlink files in the same directory... ln -s
emacs> checking for install-info... /nix/store/yag4rw6srqpl060741z50zsqpxm3886g-texinfo-7.0.3/bin/install-info
emacs> checking for gzip... /nix/store/pm6pfypn61d9n9gy19i01ak8hwc6g688-gzip-1.12/bin/gzip
emacs> checking for 'find' args to delete a file... -delete
emacs> checking for brew... no
emacs> checking for -znocombreloc... not needed
emacs> checking whether addresses are sanitized... no
emacs> checking for library containing sqrt... -lm
emacs> checking pkg-config is at least version 0.9.0... yes
emacs> checking for machine/soundcard.h... no
emacs> checking for sys/soundcard.h... yes
emacs> checking for soundcard.h... no
emacs> checking for mmsystem.h... no
emacs> checking for _oss_ioctl in -lossaudio... no
emacs> checking for alsa >= 1.0.0... no
emacs> checking for ADDR_NO_RANDOMIZE... yes
emacs> checking if Linux sysinfo may be used... yes
emacs> checking for sys/wait.h that is POSIX.1 compatible... yes
emacs> checking for net/if.h... yes
emacs> checking for ifaddrs.h... yes
emacs> checking for net/if_dl.h... no
emacs> checking for struct ifreq.ifr_flags... yes
emacs> checking for struct ifreq.ifr_hwaddr... yes
emacs> checking for struct ifreq.ifr_netmask... yes
emacs> checking for struct ifreq.ifr_broadaddr... yes
emacs> checking for struct ifreq.ifr_addr... yes
emacs> checking for struct ifreq.ifr_addr.sa_len... no
emacs> checking whether gcc understands -MMD -MF... yes
emacs> checking for X... libraries , headers
emacs> checking whether malloc is Doug Lea style... no
emacs> checking for sbrk... yes
emacs> checking for getpagesize... yes
emacs> checking for __lsan_ignore_object... no
emacs> checking for fork... yes
emacs> checking for vfork... yes
emacs> checking for fchmod... yes
emacs> checking for canonicalize_file_name... yes
emacs> checking for faccessat... yes
emacs> checking for realpath... yes
emacs> checking for lstat... yes
emacs> checking for readlinkat... yes
emacs> checking for explicit_bzero... yes
emacs> checking for memset_s... no
emacs> checking for fchmodat... yes
emacs> checking for lchmod... yes
emacs> checking for fcntl... yes
emacs> checking for fdopendir... yes
emacs> checking for fstatat... yes
emacs> checking for fsync... yes
emacs> checking for futimens... yes
emacs> checking for getrandom... yes
emacs> checking for gettimeofday... yes
emacs> checking for mkostemp... yes
emacs> checking for pipe2... yes
emacs> checking for pselect... yes
emacs> checking for pthread_sigmask... yes
emacs> checking for readlink... yes
emacs> checking for isblank... yes
emacs> checking for iswctype... yes
emacs> checking for strtoimax... yes
emacs> checking for symlink... yes
emacs> checking for localtime_r... yes
emacs> checking for timegm... yes
emacs> checking for utimensat... yes
emacs> checking for getdtablesize... yes
emacs> checking for futimes... yes
emacs> checking for futimesat... yes
emacs> checking for lutimes... yes
emacs> checking for working mmap... yes
emacs> checking for main in -lXbsd... no
emacs> checking for pthread library... none needed
emacs> checking for thread support... yes
emacs> checking whether X on GNU/Linux needs -b to link... no
emacs> checking for Xkb... yes
emacs> checking for XrmSetDatabase... yes
emacs> checking for XScreenResourceString... yes
emacs> checking for XScreenNumberOfScreen... yes
emacs> checking X11 version 6... 6 or newer
emacs> checking for librsvg-2.0 >= 2.14.0... yes
emacs> checking for getaddrinfo_a in -lanl... yes
emacs> checking for dbus-1 >= 1.0... yes
emacs> checking for dbus_watch_get_unix_fd... yes
emacs> checking for dbus_type_is_valid... yes
emacs> checking for dbus_validate_bus_name... yes
emacs> checking for dbus_validate_path... yes
emacs> checking for dbus_validate_interface... yes
emacs> checking for dbus_validate_member... yes
emacs> checking for gio-2.0 >= 2.26... yes
emacs> checking whether GSettings is in gio... yes
emacs> checking for gobject-2.0 >= 2.0... yes
emacs> checking for lgetfilecon in -lselinux... yes
emacs> checking for gnutls >= 2.12.2... yes
emacs> checking for libsystemd >= 222... yes
emacs> checking for jansson >= 2.7... yes
emacs> checking for sys/inotify.h... yes
emacs> checking for inotify_init1... yes
emacs> checking for XawScrollbarSetThumb in -lXaw3d... yes
emacs> checking for xaw3d... yes; using Lucid toolkit
emacs> checking X11 toolkit version... 6 or newer
emacs> checking for library containing XmuConvertStandardSelection... -lXmu
emacs> checking for XRenderQueryExtension in -lXrender... yes
emacs> checking for cairo >= 1.8.0... yes
emacs> checking for freetype2... yes
emacs> checking for fontconfig >= 2.2.0... yes
emacs> checking for FT_Face_GetCharVariantIndex... yes
emacs> checking for libotf... yes
emacs> checking for OTF_get_variation_glyphs in -lotf... yes
emacs> checking for m17n-flt... yes
emacs> checking for harfbuzz >= 0.9.42... yes
emacs> checking for X11/Xlib-xcb.h... yes
emacs> checking for xcb_translate_coordinates in -lxcb... yes
emacs> checking for XGetXCBConnection in -lX11-xcb... yes
emacs> checking for X11/xpm.h... yes
emacs> checking for XpmReadFileToPixmap in -lXpm... yes
emacs> checking for XpmReturnAllocPixels preprocessor define... yes
emacs> checking for jpeglib 6b or later... -ljpeg
emacs> checking for lcms2... no
emacs> checking for library containing inflateEnd... -lz
emacs> checking for dladdr... yes
emacs> checking for dlfunc... no
emacs> checking for gcc_jit_context_acquire in -lgccjit... yes
emacs> checking for libgccjit.h... yes
emacs> configure: error: The installed libgccjit failed to compile and run a test program using
emacs> the libgccjit library; see config.log for the details of the failure.
emacs> The test program can be found here:
emacs> <https://gcc.gnu.org/onlinedocs/jit/intro/tutorial01.html>.
emacs> You can try compiling it yourself to investigate the issues.
emacs> Please report the issue to your distribution if libgccjit was installed
emacs> through that.
emacs> You can find the instructions on how to compile and install libgccjit from
emacs> source on this site:
emacs> <https://gcc.gnu.org/wiki/JIT>.
error: builder for '/nix/store/0xina6mhx32aw3fv6n6drnhggrq063fq-emacs-28.2.drv' failed with exit code 1;
       last 10 log lines:
       > configure: error: The installed libgccjit failed to compile and run a test program using
       > the libgccjit library; see config.log for the details of the failure.
       > The test program can be found here:
       > <https://gcc.gnu.org/onlinedocs/jit/intro/tutorial01.html>.
       > You can try compiling it yourself to investigate the issues.
       > Please report the issue to your distribution if libgccjit was installed
       > through that.
       > You can find the instructions on how to compile and install libgccjit from
       > source on this site:
       > <https://gcc.gnu.org/wiki/JIT>.
       For full logs, run 'nix log /nix/store/0xina6mhx32aw3fv6n6drnhggrq063fq-emacs-28.2.drv'.
error: 1 dependencies of derivation '/nix/store/adxs4qgb3kcapwnc0ixwv06bd499yal1-mu-1.10.7.drv' failed to build
error: 1 dependencies of derivation '/nix/store/hs8fsbigds03240d89ignx4qgsv8x56j-lbdb-0.48.1.drv' failed to build
error: 1 dependencies of derivation '/nix/store/v2x0jbyizlpvhimcxagkj0s0i4g08m3z-review-shell.drv' failed to build

Link to currently reviewing PR:
https://github.com/NixOS/nixpkgs/pull/252562

1 package failed to build:
lbdb

2 packages built:
khard khard.dist

error: build log of '/nix/store/hs8fsbigds03240d89ignx4qgsv8x56j-lbdb-0.48.1.drv!*' is not available
error: build log of '/nix/store/dniysj82pcljw5zm1my3wx7nkrmkcwpy-lbdb-0.48.1' is not available
$ /nix/store/8vk4w1fiy2lr84xqkaqi0rjlw2s7882h-nix-2.13.5/bin/nix-shell /home/lelgenio/.cache/nixpkgs-review/pr-252562-3/shell.nix
</code>
</pre>
</details>

--- a/pkgs/applications/misc/khard/default.nix
+++ b/pkgs/applications/misc/khard/default.nix
@@ -10,9 +10,14 @@ python3.pkgs.buildPythonApplication rec {
   };
 
   SETUPTOOLS_SCM_PRETEND_VERSION = version;
-  nativeBuildInputs = [
-    python3.pkgs.setuptools-scm
+  nativeBuildInputs = with python3.pkgs; [
+    setuptools-scm
+    sphinxHook
+    sphinx-autoapi
+    sphinx-autodoc-typehints
   ];
+
+  sphinxBuilders = [ "man" ];
 
   propagatedBuildInputs = with python3.pkgs; [
     atomicwrites
@@ -40,5 +45,6 @@ python3.pkgs.buildPythonApplication rec {
     description = "Console carddav client";
     license = lib.licenses.gpl3;
     maintainers = with lib.maintainers; [ matthiasbeyer ];
+    mainProgram = "khard";
   };
 }

